semantic_search/main.py

```diff
@@ -1,13 +1,22 @@
+import os
 from operator import itemgetter
 from typing import Dict, List, Optional, Tuple, cast
 
+import faiss
+import numpy as np
 import torch
 from fastapi import FastAPI
 from pydantic import BaseSettings
 
-from semantic_search.common.util import encode_with_transformer, setup_model_and_tokenizer
-from semantic_search.schemas import Model, Query
 from semantic_search import __version__
+from semantic_search.common.util import (
+    encode_with_transformer,
+    setup_faiss_index,
+    setup_model_and_tokenizer,
+)
+from semantic_search.schemas import Model, Query
+
+os.environ["KMP_DUPLICATE_LIB_OK"] = "True"
```

Comment:
I don't know why this is necc. It is suggested by something? when running the server with FAISS. Make sure we aren't doing something dumb here.

Reply:
Here is the error without the hotfix above:

```
OMP: Error #15: Initializing libomp.dylib, but found libiomp5.dylib already initialized.
OMP: Hint This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results. For more information, please see http://openmp.llvm.org/
```

Reply:
@Anwesh1 Could you open an issue to keep track?

Reply:
#53